src/System.Text.RegularExpressions/tests/Performance/Perf.Regex.Cache.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.Xunit.Performance;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class Perf_Regex_Cache
+    {
+        private const int N = 50_000;
+        private const int UniqueRegsNum = (int)(N * 0.02);
+        private const int CacheSize = (int)(N * 0.02);
```

Comment:
if the number of unique patterns is the same as the cache size the cache will always hit, right?

Reply:
Right. Not so realistic. Don't know what's better. Make two tests: for the default cache size and for some part of unique regexes?  I'll think about it more.

Reply:
You could just make `UniqueRegsNum = (int)(N * 0.04)` or 0.06 or 0.08 or 0.10. A factor between 2 and 5 of the cache size should suffice here.